c++/src/io/OutputStream.hh

```diff
@@ -74,6 +74,11 @@ namespace orc {
     virtual bool isCompressed() const {
       return false;
     }
+
+    virtual bool isBlockCompressed() const {
```

Comment:
I would be against adding this to the `BufferedOutputStream`. We already have `isCompressed()` above, we can use static_cast to get a `CompressionStream` and call this from there. WDYT?